pytensor/tensor/rewriting/elemwise.py

```diff
@@ -742,40 +750,47 @@ def elemwise_scalar_op_has_c_code(
                     for inp in node.inputs:
                         ancestor_node = inp.owner
                         ancestor_bitflag = nodes_bitflags[ancestor_node]
-                        if ancestor_bitflag & subgraph_bitset:
+                        if (not is_ancestor) and (ancestor_bitflag & subgraph_bitset):
                             continue
                         if node in fuseable_clients.get(ancestor_node, ()):
                             heappush(
                                 fuseables_nodes_queue,
-                                (-ancestor_bitflag, ancestor_node),
+                                (
+                                    -ancestors_bitsets[ancestor_node],
+                                    ancestor_bitflag,
+                                    ancestor_node,
+                                ),
                             )
                         else:
                             # If the node is not in the ancestor's fuseable clients set, it's not fuseable with it,
                             # nor with any of the ancestor's ancestors
-                            unfuseable_ancestors_bitset |= ancestors_bitset[
+                            unfuseable_ancestors_bitset |= ancestors_bitsets[
                                 ancestor_node
                             ]
 
                     next_fuseable_clients = fuseable_clients.get(node, ())
                     for client, _ in fg_clients[node.outputs[0]]:
                         client_bitflag = nodes_bitflags[client]
-                        if client_bitflag & subgraph_bitset:
+                        if is_ancestor and (client_bitflag & subgraph_bitset):
                             continue
                         if client in next_fuseable_clients:
-                            heappush(fuseables_nodes_queue, (client_bitflag, client))
+                            heappush(
+                                fuseables_nodes_queue,
+                                (ancestors_bitsets[client], client_bitflag, client),
+                            )
                         else:
                             # If a client is not in the node's fuseable clients set, it's nto fuseable with it,
                             # nor any of its clients. But we don't need to keep track of those as any downstream
                             # client we may consider later will also depend on this unfuseable client and be rejected
                             unfuseable_clients_bitset |= client_bitflag
 
-                # Finished exploring this subgraph
-                all_subgraphs_bitset |= subgraph_bitset
-
                 if subgraph_bitset == starting_bitflag:
                     # We ended were we started, no fusion possible
                     continue
 
+                # Finished exploring this subgraph
+                all_subgraphs_bitset |= subgraph_bitset
```

Comment:
this was what was causing the scan tests to fail. If we add the subgraph_bitset of failed nodes to all_subgraphs_bitset it then causes a failure in the sorting of the subgraphs below, because it was trying to sort wrt to these ones that don't matter.

Reply:
original this variables was meant more as a "this variable was claimed or known to not be fuseable", so it would fulfill the second goal. But that info was not used anywhere in the latest iteration. Instead we only use this for the sorting bit at the end.